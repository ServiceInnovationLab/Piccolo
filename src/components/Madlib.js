import React from 'react';
import styled from 'styled-components';
import { Paragraph } from '../elements/Paragraph';

const MadlibWrapper = styled.div`
  text-align: center;

  > p {
    margin-bottom: 5px;
    font-size: 28px;
    @media(min-width: 1024px) {
      font-size: 34px;
    }
  }

  > p,
  input[type=text] {
    display: inline-block;
  }

  input[type=text] {
    width: 100px;
  }

`;

const Input = styled.input.attrs({ type: 'text' })`
  display: inline-block;
  border: none;
  border-bottom: 2px solid #000;
  font-size: 24px;
  margin: 0 10px;
  @media(min-width: 1024px) {
    font-size: 34px;
  }
`;

export const Madlib = () => (
  <MadlibWrapper>
    <Paragraph>I am</Paragraph>
    <Input />
    <Paragraph> years old, I have lived in the country for</Paragraph>
    <Input />
    <Paragraph> years.</Paragraph>
  </MadlibWrapper>
);

