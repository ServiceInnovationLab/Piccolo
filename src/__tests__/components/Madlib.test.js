import React from 'react';
import renderer from 'react-test-renderer';
import Madlib from '../../components/Madlib';
import 'jest-styled-components';

test('Madlib renders correctly', () => {
  let values = {
    age: 55,
    years_worked: 55,
    years_lived_in_country: 40,
    number_of_children: 5,
    gender: 'woman',
  };
  const tree = renderer.create(<Madlib values={values} />).toJSON();
  expect(tree).toMatchSnapshot();
});
