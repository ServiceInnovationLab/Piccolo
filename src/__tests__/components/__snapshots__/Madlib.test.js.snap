// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Madlib renders correctly 1`] = `
.c1 {
  font-size: 24px;
  color: #000;
  line-height: 1.5;
  height: auto;
}

.c0 {
  text-align: center;
}

.c0 > p {
  margin-bottom: 5px;
  font-size: 28px;
}

.c0 p,
.c0 input[type=text] {
  display: inline-block;
}

.c0 input[type=text] {
  width: 60px;
  background: #d9f8fa;
  text-align: center;
}

.c2 {
  display: inline-block;
  border: none;
  border-bottom: 2px solid #000;
  font-size: 24px;
  margin: 0 10px;
  font-weight: bold;
}

@media (min-width:1024px) {
  .c0 > p {
    font-size: 34px;
  }
}

@media (min-width:1024px) {
  .c2 {
    font-size: 34px;
  }
}

<div
  className="c0"
>
  <div>
    <p
      className="c1"
    >
      I am a
    </p>
    <input
      className="c2"
      defaultValue={55}
      readOnly={true}
      type="text"
    />
    <p
      className="c1"
    >
      year old
    </p>
    <input
      className="c2"
      defaultValue="woman"
      readOnly={true}
      style={
        Object {
          "width": "130px",
        }
      }
      type="text"
    />
    <br />
    <p
      className="c1"
    >
      who has had
    </p>
    <input
      className="c2"
      defaultValue={5}
      readOnly={true}
      type="text"
    />
    <p
      className="c1"
    >
      children.
    </p>
    <br />
    <p
      className="c1"
    >
      I have lived in the country for
    </p>
    <input
      className="c2"
      defaultValue={40}
      readOnly={true}
      type="text"
    />
    <p
      className="c1"
    >
       years, 
    </p>
    <br />
    <p
      className="c1"
    >
       and worked there for 
    </p>
    <input
      className="c2"
      defaultValue={55}
      readOnly={true}
      type="text"
    />
    <p
      className="c1"
    >
       years.
    </p>
  </div>
</div>
`;
